main/api/src/mill/api/Ctx.scala

```diff
@@ -61,7 +60,7 @@ class Ctx(
   val log: Logger,
   val home: os.Path,
   val env: Map[String, String],
-  val reporter: Int => Option[ManagedLoggedReporter],
+  val reporter: Int => Option[ZincProblemReporter],
   val bsp: BspContext
```

Comment:
@adadima Also I don't get the purpose of BspContext, is it really used somewhere ? If the purpose is for custom arguments it has only `Seq.empty` value

Reply:
The BspContext serves 2 functions: 

1. As you said, for custom compilation arguments. It's default value is Seq.empty, but if the client specifies arguments in `CompileParams` to pass on to the compilation, then the `BspContext` will incorporate that. For the run and test requests it will always be empty indeed since BSP does  not include any compile arguments information when sending these types of requests.

2. It injects a test reporter into mill's test() method on JavaModule so that we can inform BSP about test results as the tests are being run, ex: # of passed, failed, ignored tests.